src/lib/Webhook.coffee

```diff
@@ -0,0 +1,61 @@
+request = require "requestretry"
+
+DEFAULT_TIMEOUT = 10 * 1000
+DEFAULT_MAX_ATTEMPTS = 3
+
+class Webhook
+
+  constructor: (options = {}) ->
```

Comment:
I'm a little nervous about setting `this.options` here and then referring to it elsewhere. Could we unpack this a little bit? So the things that we refer to below like `username` and `attachments` we initialize here. Then below instead of referring to `@options.username`, we could refer to `this.username || options.username`.